packages/xl-ai/src/components/AIMenu/PromptSuggestionMenu.tsx

```diff
@@ -0,0 +1,135 @@
+  onManualPromptSubmit: (userPrompt: string) => void;
+  promptText?: string;
+  onPromptTextChange?: (userPrompt: string) => void;
+  placeholder?: string;
+  disabled?: boolean;
+  loading?: boolean;
+  error?: string;
+};
+
+export const PromptSuggestionMenu = (props: PromptSuggestionMenuProps) => {
+  // const dict = useAIDictionary();
+  const Components = useComponentsContext()!;
+
+  const { onManualPromptSubmit, promptText, onPromptTextChange } = props;
+
+  // Only used internal state when `props.prompText` is undefined (i.e., uncontrolled mode)
+  const [internalPromptText, setInternalPromptText] = useState<string>("");
+  const promptTextToUse = promptText || internalPromptText;
+
+  const handleEnter = useCallback(
+    async (event: KeyboardEvent) => {
+      if (event.key === "Enter") {
+        // console.log("ENTER", currentEditingPrompt);
+        onManualPromptSubmit(promptTextToUse);
+      }
+    },
+    [promptTextToUse, onManualPromptSubmit]
+  );
+
+  const handleChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const newValue = event.currentTarget.value;
+      if (onPromptTextChange) {
+        onPromptTextChange(newValue);
+      }
+
+      // Only update internal state if it's uncontrolled
+      if (promptText === undefined) {
+        setInternalPromptText(newValue);
+      }
+    },
+    [onPromptTextChange, setInternalPromptText, promptText]
+  );
+
+  const items: DefaultReactSuggestionItem[] = useMemo(() => {
+    return filterSuggestionItems(props.items, promptTextToUse);
+  }, [promptTextToUse, props.items]);
+
+  const { selectedIndex, setSelectedIndex, handler } =
+    useSuggestionMenuKeyboardHandler(items, (item) => item.onItemClick());
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      // TODO: handle backspace to close
+      if (event.key === "Enter") {
+        if (items.length > 0) {
+          handler(event);
+        } else {
+          // TODO: check focus?
+          handleEnter(event);
+        }
+      } else {
+        handler(event);
+      }
+    },
+    [handleEnter, handler, items.length]
+  );
+
+  // Resets index when items change
+  useEffect(() => {
+    setSelectedIndex(0);
+  }, [promptTextToUse, setSelectedIndex]);
+
+  return (
+    <div className={"bn-combobox"}>
+      <Components.Generic.Form.Root>
+        <Components.Generic.Form.TextInput
+          // Change the key when disabled change, so that autofocus is retriggered
+          key={"input-" + props.disabled}
+          className={"bn-combobox-input"}
+          name={"ai-prompt"}
+          variant={"large"}
+          icon={<RiSparkling2Fill />}
+          value={promptTextToUse || ""}
+          autoFocus={true}
+          placeholder={props.placeholder}
+          disabled={props.disabled}
+          onKeyDown={handleKeyDown}
+          onChange={handleChange}
+          autoComplete={"off"}
+          //  TODO: loader or error
```

Comment:
@matthewlipskicould you implement this (for mantine / ariakit / shadcn of course). My idea was to use `rightSection` for this